packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -205,10 +206,12 @@ class AndroidDevice extends Device {
         } else {
           return TargetPlatform.android_arm;
         }
+      case 'armeabi-v7a':
+        return TargetPlatform.android_arm;
       case 'x86_64':
         return TargetPlatform.android_x64;
       default:
-        return TargetPlatform.android_arm;
+        throw UnsupportedError('Unsupported Android architecture: $abi');
```

Comment:
This API seems weird - it never could throw before, and `UnsupportedError` shouldn't be caught.

I would expect this to either return `null`, or to return `TargetPlatform.unsupported` or _keep_ returning `android_arm` and use that a signal for "not supported" below.

Reply:
Keeping and returning `android_arm` means we need to keep handling that case in every switch on `TargetPlatform` throughout the tool, so I'd really like to avoid doing that.

`TargetPlatform.unsupported` might be an idea... let me try some things out.